hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/client/RMProxy.java

```diff
@@ -98,8 +98,7 @@ protected static <T> T createRMProxy(final Configuration configuration,
     YarnConfiguration conf = (configuration instanceof YarnConfiguration)
         ? (YarnConfiguration) configuration
         : new YarnConfiguration(configuration);
-    RetryPolicy retryPolicy = createRetryPolicy(conf,
-        (HAUtil.isHAEnabled(conf) || HAUtil.isFederationFailoverEnabled(conf)));
+    RetryPolicy retryPolicy = createRetryPolicy(conf, isFailoverEnabled(conf));
```

Comment:
Can we add a test?

Reply:
@goiri - I have updated the test `testProxyProvider` as part this change. Are you recommending to add unit test case for `isFailoverEnabled` method?

Reply:
Is that test covering everything we need?

Reply:
@goiri - I think the test is covering the required.